src/sdk/bucketspace_fs.js

```diff
@@ -650,6 +623,73 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
     async put_object_lock_configuration(params, object_sdk) {
         // TODO
     }
+
+    /////////////////
+    ///// UTILS /////
+    /////////////////
+
+    // TODO: move the following 3 functions - has_bucket_action_permission(), validate_fs_bucket_access(), _has_access_to_nsfs_dir()
+    // so they can be re-used
+    async has_bucket_action_permission(bucket, account, action, bucket_path = "") {
+        const account_identifier = account.name.unwrap();
+        dbg.log1('has_bucket_action_permission:', bucket.name.unwrap(), account_identifier, bucket.system_owner, bucket.owner_account);
+
+        const is_system_owner = account_identifier === bucket.system_owner.unwrap();
+
+        // If the system owner account wants to access the bucket, allow it
+        if (is_system_owner) return true;
+        const is_owner = (bucket.bucket_owner.unwrap() === account_identifier);
+        const bucket_policy = bucket.s3_policy;
+
+        if (!bucket_policy) return is_owner;
+        if (!action) {
+            throw new Error('has_bucket_action_permission: action is required');
+        }
+
+        const result = await bucket_policy_utils.has_bucket_policy_permission(
+            bucket_policy,
+            account_identifier,
+            action,
+            `arn:aws:s3:::${bucket.name.unwrap()}${bucket_path}`,
+            undefined
```

Comment:
What does it mean that we always pass undefined?
I see that in the internal function it will be `condition_fit` true, but I don't understand the whole context.

Reply:
req is needed when conditions are passed in order to run the condition's function.
see here - https://github.com/noobaa/noobaa-core/pull/7273